test/e2e/util.go

```diff
@@ -482,35 +539,49 @@ func RunRC(c *client.Client, name string, ns, image string, replicas int) error
 		return fmt.Errorf("Error creating replication controller: %v", err)
 	}
 
+	// Create a routine to query for the list of pods
+	stop := make(chan struct{})
+	go func(stop <-chan struct{}, n string, ns string, l labels.Selector) {
+		for {
+			select {
+			case <-stop:
+				return
+			default:
+				p, err := c.Pods(ns).List(l, fields.Everything())
+				if err != nil {
+					Logf("Warning: Failed to get pod list: %v", err)
+				} else {
+					podLists.Push(p.Items)
+				}
+				time.Sleep(1 * time.Second)
```

Comment:
There is a case for using watch (real time-ish updates) but there is one for not using it as well. Watch libraries are sufficiently complex (both the reflector and controller) that they might have bugs, and the test would suffer from the same. List is less likely to fail, with a 0% chance of *dropping* anything, but also going to stress out the cluster. 

It might be worth staging the list, instead of hammering the apiserver with a list on all pods.